src/main/java/com/teragrep/lsh_01/pool/ManagedRelpConnection.java

```diff
@@ -26,7 +26,7 @@
 import java.io.IOException;
 import java.util.concurrent.TimeoutException;
 
-public class ManagedRelpConnection {
+public class ManagedRelpConnection implements IManagedRelpConnection {
```

Comment:
I think this class should track the state that the connection has been connected() so there would be no need for external connect() calls?